sympy/solvers/ode/tests/test_ode.py

```diff
@@ -32,6 +32,223 @@
+        'euler_var_01': {
+            'eq': Eq(x**2*Derivative(f(x), x, x) - 2*x*Derivative(f(x), x) + 2*f(x), x**4),
+            'sol': [Eq(f(x), x*(C1 + C2*x + x**3/6))]
+        },
+
+        'euler_var_02': {
+            'eq': Eq(3*x**2*diff(f(x), x, x) + 6*x*diff(f(x), x) - 6*f(x), x**3*exp(x)),
+            'sol': [Eq(f(x), C1/x**2 + C2*x + x*exp(x)/3 - 4*exp(x)/3 + 8*exp(x)/(3*x) - 8*exp(x)/(3*x**2))]
+        },
+
+        'euler_var_03': {
+            'eq': Eq(x**2*Derivative(f(x), x, x) - 2*x*Derivative(f(x), x) + 2*f(x), x**4*exp(x)),
+            'sol':  [Eq(f(x), x*(C1 + C2*x + x*exp(x) - 2*exp(x)))]
+        },
+
+        'euler_var_04': {
+            'eq': x**2*Derivative(f(x), x, x) - 2*x*Derivative(f(x), x) + 2*f(x) - log(x),
+            'sol': [Eq(f(x), C1*x + C2*x**2 + log(x)/2 + Rational(3, 4))]
+        },
+
+        'euler_var_05': {
+            'eq': -exp(x) + (x*Derivative(f(x), (x, 2)) + Derivative(f(x), x))/x,
+            'sol': [Eq(f(x), C1 + C2*log(x) + exp(x) - Ei(x))]
+        },
+    }
+}
+
+all_hints = [
+    "factorable",
+    "nth_algebraic",
+    "separable",
+    "1st_exact",
+    "1st_linear",
+    "Bernoulli",
+    "Riccati_special_minus2",
+    "1st_homogeneous_coeff_best",
+    "1st_homogeneous_coeff_subs_indep_div_dep",
+    "1st_homogeneous_coeff_subs_dep_div_indep",
+    "almost_linear",
+    "linear_coefficients",
+    "separable_reduced",
+    "1st_power_series",
+    "lie_group",
+    "nth_linear_constant_coeff_homogeneous",
+    "nth_linear_euler_eq_homogeneous",
+    "nth_linear_constant_coeff_undetermined_coefficients",
+    "nth_linear_euler_eq_nonhomogeneous_undetermined_coefficients",
+    "nth_linear_constant_coeff_variation_of_parameters",
+    "nth_linear_euler_eq_nonhomogeneous_variation_of_parameters",
+    "Liouville",
+    "2nd_linear_airy",
+    "2nd_linear_bessel",
+    "2nd_hypergeometric",
+    "2nd_hypergeometric_Integral",
+    "nth_order_reducible",
+    "2nd_power_series_ordinary",
+    "2nd_power_series_regular",
+    "nth_algebraic_Integral",
+    "separable_Integral",
+    "1st_exact_Integral",
+    "1st_linear_Integral",
+    "Bernoulli_Integral",
+    "1st_homogeneous_coeff_subs_indep_div_dep_Integral",
+    "1st_homogeneous_coeff_subs_dep_div_indep_Integral",
+    "almost_linear_Integral",
+    "linear_coefficients_Integral",
+    "separable_reduced_Integral",
+    "nth_linear_constant_coeff_variation_of_parameters_Integral",
+    "nth_linear_euler_eq_nonhomogeneous_variation_of_parameters_Integral",
+    "Liouville_Integral",
+       ]
+all_solvers = [ode_sol_euler_homogeneous, ode_sol_euler_undetermined_coeff, ode_sol_euler_var_para]
+all_examples = []
+
+for solver in all_solvers:
+    for example in solver['examples']:
+        all_examples.append(solver['examples'][example]['eq'])
+
+
+def _test_all_hints():
+    for our_hint in all_hints:
+        for ode_example in all_examples:
+            assert _test_for_particular_hint(our_hint, ode_example)
+
 
```

Comment:
@oscarbenjamin does this approach look good for collecting all the ode examples and testing against every hint?

Reply:
The combined data structure should store all of the data for each example: the ODE, the solution and the associated hint.

Reply:
Wouldn't that be kind of redundant? As currently there are dictionaries of solvers having hint and examples as keys and examples include equation and solution only.